ingeniamotion/fsoe.py

```diff
@@ -169,16 +192,40 @@ def wait_for_data_state(self, timeout: Optional[float] = None) -> None:
             if timeout and (init_time + timeout) < time.time():
                 raise IMTimeoutError("The FSoE Master did not reach the Data state")
 
-    @staticmethod
-    def _saco_phase_1_dictionary() -> "Dictionary":
+    def _saco_phase_1_dictionary(self) -> "Dictionary":
         """Get the SaCo phase 1 dictionary instance"""
         sto_command_dict_item = DictionaryItemInputOutput(
-            key=0x040,
-            name="STO_COMMAND",
+            key=self.STO_COMMAND_KEY,
+            name=self.STO_COMMAND_UID,
+            data_type=DictionaryItem.DataTypes.BOOL,
+            fail_safe_input_value=True,
+        )
+        ss1_command_dict_item = DictionaryItemInputOutput(
+            key=self.SS1_COMMAND_KEY,
+            name=self.SS1_COMMAND_UID,
             data_type=DictionaryItem.DataTypes.BOOL,
             fail_safe_input_value=True,
         )
-        return Dictionary([sto_command_dict_item])
+        sbc_command_dict_item = DictionaryItemInputOutput(
+            key=self.SBC_COMMAND_KEY,
+            name=self.SBC_COMMAND_UID,
+            data_type=DictionaryItem.DataTypes.BOOL,
+            fail_safe_input_value=False,
+        )
+        safe_input_dict_item = DictionaryItemInput(
+            key=self.SAFE_INPUTS_KEY,
```

Comment:
I think we should also have an api for the new elements. I think it has been requested at least for safe inputs